File: app/src/main/java/net/squanchy/search/SearchInjector.java
package net.squanchy.search;

import net.squanchy.injection.ActivityContextModule;
import net.squanchy.injection.ApplicationInjector;
import net.squanchy.navigation.NavigationModule;

final class SearchInjector {

    private SearchInjector() {
        // no instances
    }

    public static SearchComponent obtain(SearchActivity activity) {
        return DaggerSearchComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(activity))
                .searchModule(new SearchModule())
                .activityContextModule(new ActivityContextModule(activity))
                .navigationModule(new NavigationModule())
                .build();
    }
}

File: app/src/main/java/net/squanchy/search/SearchInjector.kt
package net.squanchy.search

import net.squanchy.injection.ActivityContextModule
import net.squanchy.injection.applicationComponent
import net.squanchy.navigation.NavigationModule

internal fun searchComponent(activity: SearchActivity) =
    DaggerSearchComponent.builder()
        .applicationComponent(activity.applicationComponent)
        .searchModule(SearchModule())
        .activityContextModule(ActivityContextModule(activity))
        .navigationModule(NavigationModule())
        .build()
